src/Microsoft.ML.StandardLearners/Standard/LinearClassificationTrainer.cs

```diff
@@ -1578,8 +1589,49 @@ internal void Check(IHostEnvironment env)
 
         public override TrainerInfo Info { get; }
 
-        public StochasticGradientDescentClassificationTrainer(IHostEnvironment env, Arguments args)
-            : base(env, LoadNameValue)
+        /// <summary>
+        /// Initializes a new instance of <see cref="StochasticGradientDescentClassificationTrainer"/>
+        /// </summary>
+        /// <param name="env">The environment to use.</param>
+        /// <param name="featureColumn">The name of the feature column.</param>
+        /// <param name="labelColumn">The name of the label column.</param>
+        /// <param name="weightColumn">The name for the example weight column.</param>
+        /// <param name="maxIterations">The maximum number of iterations; set to 1 to simulate online learning.</param>
+        /// <param name="initLearningRate">The initial Learning Rate used by SGD.</param>
+        /// <param name="l2Weight">The L2 regularizer constant.</param>
+        /// <param name="advancedSettings">A delegate to apply all the advanced arguments to the algorithm.</param>
+        public StochasticGradientDescentClassificationTrainer(IHostEnvironment env, string featureColumn, string labelColumn, string weightColumn = null,
+            int maxIterations = Arguments.Defaults.MaxIterations,
+            double initLearningRate = Arguments.Defaults.InitLearningRate,
+            float l2Weight = Arguments.Defaults.L2Weight,
+            Action<Arguments> advancedSettings = null)
+            : base(env, featureColumn, TrainerUtils.MakeBoolScalarLabel(labelColumn), weightColumn)
+        {
+            Host.CheckNonEmpty(featureColumn, nameof(featureColumn));
+            Host.CheckNonEmpty(labelColumn, nameof(labelColumn));
+
+            _args = new Arguments();
+            advancedSettings?.Invoke(_args);
+
+            // Apply the advanced args, if the user supplied any.
+            _args.FeatureColumn = featureColumn;
+            _args.LabelColumn = labelColumn;
+            _args.WeightColumn = weightColumn;
+            _args.MaxIterations = maxIterations;
+            _args.InitLearningRate = initLearningRate;
+            _args.L2Weight = l2Weight;
```

Comment:
for fast trees, i am printing a warning to the user to let them know that the values of the arguments are being overridden. Since we give them two places to set them, to avoid, feel like we should warn. 

       [https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.FastTree/FastTree.cs#L169](https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.FastTree/FastTree.cs#L169)


Also:
[https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.FastTree/FastTree.cs#L104](https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.FastTree/FastTree.cs#L104)